### ISSUE TITLE
Migrate to GatsbyJS

### ISSUE THREAD

**Issue by juanpicado** [verdaccio/website#248]
We have been using https://github.com/facebook/docusaurus for a very long time, I'm really proud of how they have managed to push this project in a key area, documentation.

Docusaurus v2 still does not support translations and is hard to customize. I've been working in parallel to create a dummy migration to GatsbyJS (I'll move the repo to this organization soon), if you are willing to contribute, the stack is the following.

* Typescript
* Gatsby.js
* Material-UI

Just ping me, I'll be working on this the next months.

**Comment by DanielRuf** [comment]
Well, I'm not sure if we really want to replace docusaurus with Gatsby - as Gatsby is primarily for blogging and websites and not technical documentations with versioned docs.

I would be interested to help there but we need a scalable and good solution and should keep all the things that the current docs provide now - and the same URLs.

**Comment by juanpicado** [comment]
I removed the Versioning, it is just not scalable, the website was around 500mb and timeouts were making builds around 15 min locally and Docuv1 just cannot handle that if I enable languages. 

I have already a tiny PoC running already.

**Comment by DanielRuf** [comment]
Thanks for the additional information. Ok then it makes sense to have a website and documentation only for the latest version. But where can we point users to for old docs when we have no docsets or similar solutions?

**Comment by juanpicado** [comment]
>  But where can we point users to for old docs when we have no docsets or similar solutions?

Docs won't change, same files, same URL, same translations. Is that what you are referring for?

**Comment by DanielRuf** [comment]
Well, for which parts do we use Gatsby then when you write

> I removed the Versioning, it is just not scalable, the website was around 500mb and timeouts were making builds fail, builds also was taking 15 min locally and Docuv1 just cannot handle that if I enable languages. Crowdin translations were messed up with docu1.

> Docusaurus v2 still does not support translations and is hard to customize. I've been working in parallel to create a dummy migration to GatsbyJS

So you mean the already generated pages are copied to the Gatsby project and we stop using docusaurus as compiler / generator then but use the output that we have so far? That is not quite clear to me at the moment.

**Comment by sergiohgz** [comment]
IMHO, we have to discuss first what are our requirements for a site.

I think our docs are really good, but the space and time they take to build are really awful. 

I want to have a tool that makes things easy to maintain. Docusaurus and Gatsby are really great for that, but there are some requirements both should handle:

- Site template should be unique, not replicated across different versions 
- Main page and sections across site should be translated 
- Docs content should be versioned and translated

I would say to not maintain all the versions we have, that could be too much for the scope, but keeping a 3.x branch and current branch, should all what we need

**Comment by juanpicado** [comment]
> So you mean the already generated pages are copied to the Gatsby project and we stop using docusaurus as compiler / generator then but use the output that we have so far? 

Yes, the idea is stoping to use Docusaurus. Gatsby is way more flexible. 

Let me explain the current workflow and issues. 

Crowdin currently picks from a folder the original source and eventually PR the translation changes via PR every time anything is being translated (even a small word). This is quite annoying since I have to merge almost every week (or day) and has to be squash, otherwise crowding creates a commit for each (language/file/line) translated, which is overwhelming IHMO.

Docusaurus is great, but the v1 has a big problem, cannot be extended easily, no plugins, SEO is horrible and super slow as I described above for our current requirements. React is not flexible at all since I cannot import modules and not even require external files. V2 still in development and seems to be quite similar. Obviously is plug and play and that's awesome, but the blog system is quite limited, I'd like to be able to use plugins (or build my own ones) to extend functionality and try new things.

My goal is to increase visits, we have around 10k per month, not bad, but only generates 40$ monthly in ads due the bad SEO.

My plan by far is to have a similar stack as UI, same libraries, and maybe a similar style using material-ui, that will help on contributions since it would be similar. Furthermore, use the tooling created by @sergiohgz in monorepo which saves tons of time, same style code across all projects, and easy to be deployed (to our CDN) <-- this is actually working for this repo and the PoC, check the GH actions.

Here an example of the PoC https://alpha.verdaccio.dev/  yeah, looks awful, just small example to play around with ideas.

By far has been easy, I have experience with Gatsby.js and I'm front-end :) so, I', open to working actively on this. My goal is to increase visits and developer experience contributions. I don't expect much changes on this project aside small adjustments once is finished, the current development is like 20%, most of the plugins are already done, design (we have one proposal), and to get this done might be in a few months after summer. There is no rush, but this is something I've been thinking for a year by now.

**Comment by juanpicado** [comment]
Follow up on `5.x`
